plugins/workspace-plugin/src/workspace-projects-manager.ts

```diff
@@ -103,6 +108,15 @@ export class WorkspaceProjectsManager {
     onDidCloneSourcesEmitter.fire();
   }
 
+  // back compatibility for single-root workspaces
+  // we need it to support workspaces wich were created before switching multi-root mode to ON by default
+  protected ensureWorkspaceFoldersFor(projects: che.devfile.DevfileProject[]): void {
```

Comment:
I think this method is misnamed: as I read the code, it filters out projects that do _not_ exist. If I read `ensureWorkspaceFolders`, I would expect the missing folders to be created. Maybe just inline this method: it's only a single statement.

Reply:
thank you, will update my PR soon!